Lab 8.ipynb

```diff
@@ -0,0 +1,652 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "fig, ax = plt.subplots(3, 1, figsize = (8,15))\n",
+    "plt.close()\n",
+    "fig2, ax2 = plt.subplots(1, 1)\n",
+    "\n",
+    "n_higgs = (100 / (100 + 20000)) * (len(high_lumi['mass']) / len(higgs['mass']))\n",
+    "n_qcd = (20000 / (100 + 20000)) * (len(high_lumi['mass']) / len(qcd['mass']))\n",
+    "\n",
+    "qcd_weight = np.array([n_qcd] * len(qcd['mass']))\n",
+    "higgs_weight = np.array([n_higgs] * len(higgs['mass']))\n",
+    "\n",
+    "mass_lower = qcd['mass'].min()\n",
+    "mass_upper = qcd['mass'].max()\n",
+    "\n",
+    "lower_val = qcd['d2'].min()\n",
+    "upper_val = qcd['d2'].max()\n",
+    "\n",
+    "bin_num = 100\n",
+    "\n",
+    "hist_qcd = ax[0].hist2d(qcd['mass'], qcd['d2'], bins = [95, bin_num], range = [[mass_lower, mass_upper], [lower_val, upper_val]], density = False, weights = qcd_weight)\n",
```

Comment:
Would be clearer if you defined variables instead of just having magic numbers for the bin size